### DISCUSSION TITLE
Sync is very slow

### DISCUSSION THREAD

**Discussion by dbogdanov** [gri38/django-joplin_vieweb#42] [General]
I am stuck on synchronization importing a large number of notes (>1K) through the web UI for multiple hours. Is Joplin Data API slow? From the log output, it appears that a single query is done precisely every three seconds.
```
[27/Jul/2021 23:58:47] "GET /joplin/sync/ HTTP/1.1" 200 17
[27/Jul/2021 23:58:50] "GET /joplin/sync/ HTTP/1.1" 200 17
[27/Jul/2021 23:58:53] "GET /joplin/sync/ HTTP/1.1" 200 17
[27/Jul/2021 23:58:56] "GET /joplin/sync/ HTTP/1.1" 200 17
```

**Comment by gri38** [comment]
Hello.
The 3 seconds is a polling for GUI to check if sync is finished, not the true query.
I would advise to do a first synchro from Joplin terminal, that's what I do when I install joplin-vieweb on a server: `joplin sync`

I want to give some progress information in the GUI (cf. issue #36).

For information, when the GUI asks a sync, the django server does a system call to `joplin`. So joplin must be a command accessible from the path (that should be the case if you installed according to this page, especially: sudo ln -s ~/.joplin-bin/bin/joplin /usr/bin/joplin).

Today I don't check if this call succeeds: I should (I will in #36).

Let me know if `joplin sync` helps you, and if `joplin` was "in you path" or not.

**Comment by dbogdanov** [comment]
Hi @gri38,

The initial problem was that opening the web page, I get 
```
Error while retrieving notebooks.
Please try again 
```
even though I did `joplin sync` in advance.

Therefore, I tried to run synchronization from the web page.

I validated that the `joplin` path is corrent.

Perhaps, something wrong with the configuration? I used the default settings from the README:

```
# Joplin variables
JOPLIN_SERVER_URL="http://127.0.0.1"
JOPLIN_SERVER_PORT=41184
JOPLIN_SERVER_TOKEN="1234567890987654321"
JOPLIN_RESSOURCES_PATH="/home/dbogdanov/.config/joplin/resources/"
JOPLIN_LOGIN_REQUIRED=True # set to True only if you require a logged user for accessing the notes
JOPLIN_SYNC_PERIOD_S=86400 # once a day
JOPLIN_SYNC_INFO_FILE="/home/dbogdanov/.config/joplin/joplin_vieweb_sync_info"
```

`/home/dbogdanov/.config/joplin/` is the folder for Joplin CLI.

**Comment by gri38** [comment]
Hi.
Is 1234567890987654321 is your joplin terminal token ?
It seems you have to set it yourself, and each time ! cf. [my question without answer on joplin forum](https://discourse.joplinapp.org/t/joplin-termin-token-deleted/17765).

So: have you set the joplin token for joplin terminal before launching joplin terminal.
You will find how to do [here](https://github.com/gri38/django-joplin_vieweb/wiki/Joplin-Terminal-Setup).

**Comment by dbogdanov** [comment]
This was the issue, thank you! I've set the token following your documentation and got pass  the `Error while retrieving notebooks` error.

However, now I have empty notebooks ("No note here...") even though `joplin sync` was run before.

**Comment by gri38** [comment]
Great.

About "no notes here..." : this message is displayed when the system succeed in gettings notes of the selected notebook, and there is no note for that notenook.
Strange.
Are you sure you have notes on that folder ?

If yes: you can have a look at the output of joplin server (I don't know how did you choose to launch it: I do in a [screen](https://www.gnu.org/software/screen/), so I can see the accesses: have  alook to the output, it can give a tip.

And also you can have a look to daphne logs (or django logs, I don't know your setup).
For instance, when I start joplin-vieweb on a browser, and I click a notebook, I cann see those logs:

127.0.0.1:48464 - - [28/Jul/2021:15:29:51] "GET /joplin/" 200 3357
127.0.0.1:48468 - - [28/Jul/2021:15:29:55] "GET /joplin/notebooks/" **200** 6899
127.0.0.1:48470 - - [28/Jul/2021:15:29:55] "GET /joplin/tags/" 200 2502
127.0.0.1:48476 - - [28/Jul/2021:15:29:58] "GET /joplin/notebooks/390d5b0e33444ad2ad42d4d8f76d5061/" **200** 193

The 2nd line (GET /joplin/notebooks/) is the notebooks retrieval: it succeeds with a **200** code: success in getting all the notebooks.
Then when I click on a notebook, I can see the request in the 4th line (GET /joplin/notebooks/390d5b0e33444ad2ad42d4d8f76d5061/) which also ens with a **200** code: all the notes have been retrieved.

Let me know about you

**Comment by gri38** [comment]
hi @dbogdanov.
I moved this issue to a discussion, if we encounter other issues than #36 we can open them.
I'm interested (if you continue using joplin-vieweb) in your feedback about the responsivness: you have big amount of notes (> 1k): it's not my case. 
And thus, I don't know if the notebooks loading is fast or not...

Don't hesitate to report any difficulty you encounter. For instance, the periodic sync written in urls.py make the commands `python migrate.py collectstatic` and `python manage.py migrate` blocking: we must stop them with a ctrl+C: I noticed that a few days ago when I reinstall in a new SD card on my raspberry: I'll have to change...
